src/metaopt/io/database/mongodb.py

```diff
@@ -26,72 +26,77 @@ def __init__(self, dbhost='localhost', dbname='metaopt', dbport=27017, \
         self.dbname = str(dbname)
         self.username = str(username)
         self.password = str(password)
+        self.dbport = str(dbport)
 
-        # we might have a different port for the database port
-        # so here we do a check if the port is there
+        # try to initialize connection
+        # setting this as a function allows to make sure reconnection
+        # is properly done in the case of an unreliable network
+        self.uri = property(self.get_uri, self.set_uri, \
+                            None, 'This is the URI string of MongoDB')
+        self.conn = self.initiate_connection()
+        self.db = self.conn[self.dbname]
 
-        self.dbport = str(dbport)
+    def get_uri(self):
+        return self._uri
 
+    def set_uri(self, uri):
         # set the connection string
         # this should be mongodb://mongodb0.example.net:27017
         # according to MongoDB documentation
         # https://docs.mongodb.com/getting-started/python/client/
+        # simplest URI string is as followed
+        # self.uri = 'mongodb://' + self.dbhost + ':' + self.dbport
 
-        self.uri = 'mongodb://' + self.dbhost + ':' + self.dbport
-        #self.uri = "mongodb://%s:%s@%s:%s" % (quote_plus(self.username), \
-        #                                      quote_plus(self.password), \
-        #                                     self.dbhost, self.dbport)
-        # try to initialize connection
-        # setting this as a function allows to make sure reconnection
-        # is properly done in the case of an unreliable network
-
-        self.conn = self.initiate_connection()
-        self.db = self.conn[self.dbname]
+        uri = "mongodb://%s:%s@%s:%s" % (quote_plus(self._username), \
+                                         quote_plus(self._password), \
+                                         self._dbhost, self._dbport)
+        self._uri = uri
 
     def initiate_connection(self):
         try:
             conn = MongoClient(self.uri)
             self.conn = conn
-        except pymongo.errors.ConnectionFailure:
-            raise Exception("Could not connect to server")
+        except pymongo.errors.ConnectionFailure as e:
+            raise Exception("Could not connect to server: %s", e)
         # return connection string for checking purposes
         return conn
 
     def check_connection(self):
         if self.db:
-            #print("Connection Successful!")
+            print("Connection is successful!")
             return True
         else:
-            #print("Please check your connection")
+            print("Please check your connection")
             return False
 
     def close_connection(self):
         try:
             close = self.conn.close()
-        except pymongo.errors.ConnectionFailure:
-            raise Exception("Could not close socket connection")
+        except pymongo.errors.ConnectionFailure as e:
```

Comment:
I was not saying we should do this. I was just wondering what kind of information is passed through ConnectionFailure. Is there any relevant information that we should not lose? Additionally, I believe raising Exception like this might break the stack trace. Look at [this](https://github.com/mila-udem/blocks/blob/master/blocks/utils/utils.py#L32) for instance. For simplicity, could the ConnectionFailure error message be good enough to just let it pass with no try-except clause?